test/e2e/storage/provision.go

```diff
@@ -145,4 +147,117 @@ var _ = ginkgo.Describe("provision volumes with different volume modes from volu
 			framework.Failf("expected failure message [%s] not parsed in event list for PVC %s/%s", volumeModeConversionFailureMessage, pvc2.Namespace, pvc2.Name)
 		}
 	})
+
```

Comment:
Can you switch the order of PVC and VolumeSnapshot deletion and have VolumeSnapshot deleted first?  Some storage systems can't delete volumes when there are snapshots on them.

Reply:
You can submit a followup PR to make this change as it is not introduced by this PR.